frameworks/C/octane/src/program.cpp

```diff
@@ -0,0 +1,49 @@
+#include <stdio.h>
+#include <uv.h>
+#include <stdbool.h>
+#include <octane.h>
+#include "responders/sds_responder.hpp"
+
+void timer_callback(uv_timer_t* timer);
+
+void on_request(http_connection* connection, http_request** requests, int number_of_requests);
+
+char* current_time;
+uv_timer_t timer;
+
+int main(int argc, char *argv[]) {
+    http_listener* listener = new_http_listener();
+    uv_timer_init(listener->loop, &timer);
+    uv_timer_start(&timer, timer_callback, 0, 500);
+
+    begin_listening(listener, "0.0.0.0", 8000, false, 4, 128, NULL, NULL, NULL, on_request);
+
+    printf("Listening...\n");
+}
+
+void on_request(http_connection* connection, http_request** requests, int number_of_requests) {
+    write_batch* batch = create_write_batch(number_of_requests);
+
+    for (int i=0; i<number_of_requests; i++) {
+        if (requests[i]->path[1] == 'p') {
+            create_plaintext_response_sds(batch);
+        } else if (requests[i]->path[1] == 'j') {
```

Comment:
@nbrady-techempower No worries, I'm under the microscope now haha :)

No, it doesn't give me any optimizations for setting the path. Octane uses a lot of zero-copy optimizations (like many of the top results in the benchmarks) but that's totally valid. All that means is Octane reduces the amount of memory copying it's doing and reuses the original buffer used for receiving from the socket by just pointing to positions in the buffer.

I'll probably add more extensive routing at some point but I think it's going to be an optional feature because different people have different routing requirements. I'm trying to build something that has layers where people can opt in to what they want or replace what they don't want.

I actually think a benchmark with a few hundred routes would be a good addition. I suspect various frameworks have much different performance characteristics based on their routing performance that doesn't show up in the plaintext and json benchmarks.

Frameworks who don't have routing would be excluded from that list.